src/hip/gpu_get2e_subs_hrr.h

```diff
@@ -0,0 +1,1899 @@
+                                                   QUICKDouble RDx,QUICKDouble RDy,QUICKDouble RDz)
+{
+    QUICKDouble Y = LOCSTORE(store,  \
+                         (int) LOC3(devTrans, LOC2(devSim.KLMN,0,III-1,3,devSim.nbasis) + LOC2(devSim.KLMN,0,JJJ-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,1,III-1,3,devSim.nbasis) + LOC2(devSim.KLMN,1,JJJ-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,2,III-1,3,devSim.nbasis) + LOC2(devSim.KLMN,2,JJJ-1,3,devSim.nbasis), TRANSDIM, TRANSDIM, TRANSDIM)-1,
+                         (int) LOC3(devTrans, LOC2(devSim.KLMN,0,KKK-1,3,devSim.nbasis) + LOC2(devSim.KLMN,0,LLL-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,1,KKK-1,3,devSim.nbasis) + LOC2(devSim.KLMN,1,LLL-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,2,KKK-1,3,devSim.nbasis) + LOC2(devSim.KLMN,2,LLL-1,3,devSim.nbasis), TRANSDIM, TRANSDIM, TRANSDIM)-1 , \
+                         STOREDIM, STOREDIM) * devSim.cons[III-1] * devSim.cons[JJJ-1] * devSim.cons[KKK-1] * devSim.cons[LLL-1];
+    
+    return Y;
+}
+
+__device__ __forceinline__ QUICKDouble hrrwhole2_10(int I, int J, int K, int L, \
+                                                    int III, int JJJ, int KKK, int LLL, int IJKLTYPE, QUICKDouble* store, \
+                                                    QUICKDouble RAx,QUICKDouble RAy,QUICKDouble RAz, \
+                                                    QUICKDouble RBx,QUICKDouble RBy,QUICKDouble RBz, \
+                                                    QUICKDouble RCx,QUICKDouble RCy,QUICKDouble RCz, \
+                                                    QUICKDouble RDx,QUICKDouble RDy,QUICKDouble RDz)
+{
+    QUICKDouble Y = LOCSTORE(store,  \
+                         (int) LOC3(devTrans, LOC2(devSim.KLMN,0,III-1,3,devSim.nbasis) + LOC2(devSim.KLMN,0,JJJ-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,1,III-1,3,devSim.nbasis) + LOC2(devSim.KLMN,1,JJJ-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,2,III-1,3,devSim.nbasis) + LOC2(devSim.KLMN,2,JJJ-1,3,devSim.nbasis), TRANSDIM, TRANSDIM, TRANSDIM)-1,
+                         (int) LOC3(devTrans, LOC2(devSim.KLMN,0,KKK-1,3,devSim.nbasis) + LOC2(devSim.KLMN,0,LLL-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,1,KKK-1,3,devSim.nbasis) + LOC2(devSim.KLMN,1,LLL-1,3,devSim.nbasis), \
+                                    LOC2(devSim.KLMN,2,KKK-1,3,devSim.nbasis) + LOC2(devSim.KLMN,2,LLL-1,3,devSim.nbasis), TRANSDIM, TRANSDIM, TRANSDIM)-1 , \
+                         STOREDIM, STOREDIM) * devSim.cons[III-1] * devSim.cons[JJJ-1] * devSim.cons[KKK-1] * devSim.cons[LLL-1];
+    
+    return Y;
+}
+
+
+__device__ __forceinline__ int lefthrr1(QUICKDouble RAx, QUICKDouble RAy, QUICKDouble RAz,
```

Comment:
Should we declare the variables that do not change with `const` type qualifier (e.g. `const QUICKDouble RAx, ` etc)? In principle it can help the compiler, although I do not know if it affects performance in this case. It's not consistent here with the corresponding file in the `cuda` directory. Neither of which use `const` for all functions.

Reply:
You are absolutely right. It helps, thats why I added them in _sp and _spd functions. The ones I didnt touch are related to f kernels. I will make relevant changes in upcoming high angular momentum integral PR. Because if we decide to combine the f kernels, some of these functions may not useful at all.